traits/tests/test_extended_trait_change.py

```diff
@@ -127,39 +137,82 @@ class Instance1(HasTraits):
 
     @on_trait_change("ref.value")
     def arg_check0(self):
-        self.calls += 1
+        self.calls[0] += 1
 
     @on_trait_change("ref.value")
     def arg_check1(self, new):
-        self.calls += 1
+        self.calls[1] += 1
         self.tc.assertEqual(new, self.dst_new)
 
     @on_trait_change("ref.value")
     def arg_check2(self, name, new):
-        self.calls += 1
+        self.calls[2] += 1
         self.tc.assertEqual(name, self.dst_name)
         self.tc.assertEqual(new, self.dst_new)
 
     @on_trait_change("ref.value")
     def arg_check3(self, object, name, new):
-        self.calls += 1
+        self.calls[3] += 1
         self.tc.assertIs(object, self.exp_object)
         self.tc.assertEqual(name, self.exp_name)
         self.tc.assertEqual(new, self.exp_new)
 
     @on_trait_change("ref.value")
     def arg_check4(self, object, name, old, new):
-        self.calls += 1
+        self.calls[4] += 1
         self.tc.assertIs(object, self.exp_object)
         self.tc.assertEqual(name, self.exp_name)
         self.tc.assertEqual(old, self.exp_old)
         self.tc.assertEqual(new, self.exp_new)
 
 
+class Instance2(Instance1):
+
+    ref = Instance(ArgCheckList, ())
+
+
+class Instance3(Instance2):
+
+    @on_trait_change("ref.value[]")
+    def arg_check0(self):
+        self.calls[0] += 1
+
+    @on_trait_change("ref.value[]")
+    def arg_check1(self, new):
+        self.calls[1] += 1
+        self.tc.assertEqual(new, self.dst_new)
```

Comment:
Any idea what `dst` is supposed to stand for?

Reply:
I am guessing "dst" is destination and "src" is source, but I don't have the mental model for _why_ those terms.  It clearly made sense to someone (probably Dave Morrill) but there's no deeper explanation that I have found in the documentation that explains the terms.